server/src/routes/assignments.ts

```diff
@@ -0,0 +1,104 @@
+import express from "express";
+
+import { User, UserRole, AssignmentStatus } from "@prisma/client";
+
+import { asyncHandler } from "../utils/asyncHandler";
+import { prisma } from "../common";
+
+export const assignmentRoutes = express.Router();
+
+assignmentRoutes.route("/").get(
+  asyncHandler(async (req, res) => {
+    const { hackathon, expo, round, categoryGroup } = req.query;
+    const filter: any = {};
+    if (hackathon || expo || round) {
+      filter.project = {};
+    }
+
+    if (hackathon !== undefined) {
+      const hackathonId: number = parseInt(hackathon as string);
+      filter.project.categories = { every: { hackathonId } };
+    }
+
+    if (expo !== undefined) {
+      const expoNumber: number = parseInt(expo as string);
+      filter.project.expo = expoNumber;
+    }
+
+    if (round !== undefined) {
+      const roundNumber: number = parseInt(round as string);
+      filter.project.round = roundNumber;
+    }
+
+    if (categoryGroup !== undefined) {
+      const categoryGroupId: number = parseInt(categoryGroup as string);
+      filter.user = {
+        categoryGroupId,
+      };
+    }
+
+    const assignments = await prisma.assignment.findMany({
+      where: filter,
+    });
+    res.status(200).json(assignments);
+  })
+);
+
+assignmentRoutes.route("/").post(
+  asyncHandler(async (req, res) => {
+    const user: User = req.body.user as User;
+    const projectId: number = parseInt(req.body.project.id);
+    const duplicateFilter: any = {};
+    const multipleProjectFilter: any = {};
+    if (user.role !== UserRole.JUDGE && user.role !== UserRole.JUDGE_AND_SPONSOR) {
+      res.status(500).json({ error: "User is not a judge" });
+      return;
+    }
+
+    duplicateFilter.projectId = projectId;
+    duplicateFilter.userId = user.id;
+
+    multipleProjectFilter.userId = user.id;
+    multipleProjectFilter.status = AssignmentStatus.STARTED;
+
+    const checkAssignment = await prisma.assignment.findMany({
+      where: {
+        OR: [duplicateFilter, multipleProjectFilter],
+      },
+    });
+
+    if (checkAssignment.length !== 0) {
+      res
+        .status(500)
+        .json({
+          error: "Judge already has a project started or project assignment is a duplicate",
+        });
+      return;
+    }
+
+    const createdAssignment = await prisma.assignment.create({
+      data: req.body.data,
```

Comment:
Had to use body.data here since if we have the user and project within the body prisma doesn't create it properly. The alternative is to write a function like `const picked = (({ a, c }) => ({ a, c }))(object);` to get a subset of the request body. If we choose this approach we should probably refactor all of the other POST/PATCH endpoints to use this convention